crates/typst-library/src/compute/data.rs

```diff
@@ -119,11 +123,44 @@ pub fn csv(
     let Spanned { v: path, span } = path;
     let id = vm.location().join(&path).at(span)?;
     let data = vm.world().file(id).at(span)?;
+    csv_decode(Spanned::new(Readable::Bytes(data), span), delimiter)
+}
 
+/// Reads structured data from a CSV string/bytes.
+///
+/// The CSV string/bytes will be read and parsed into a 2-dimensional array of strings:
+/// Each row in the CSV file will be represented as an array of strings, and all
+/// rows will be collected into a single array. Header rows will not be
+/// stripped.
+///
+/// ## Example { #example }
+/// ```example
+/// #let results = csv.decode(read("data.csv"))
+///
+/// #table(
+///   columns: 2,
+///   [*Condition*], [*Result*],
+///   ..results.flatten(),
+/// )
```

Comment:
General comment: ideally the decode examples would be distinct from the normal ones and show cases where it's actually useful opposed to the normal one

Reply:
How would you showcase that?
- Using a plugin call and interpreting the result als csv
- a string manipulation?

What do you have in mind?

Reply:
I have removed the example for now; maybe we find a good one later

Reply:
That's fine.